internal/cmd/skupper/site/site.go

```diff
@@ -18,15 +31,34 @@ func NewCmdSite() *cobra.Command {
 skupper site get my-site`,
 	}
 
-	siteCreateCommand := kube.NewCmdSiteCreate()
-	siteStatusCommand := kube.NewCmdSiteStatus()
-	siteUpdateCommand := kube.NewCmdSiteUpdate()
-	siteDeleteCommand := kube.NewCmdSiteDelete()
+	switch selectedPlatform {
+	case "kubernetes":
+		return AddKubeSiteCommands(cmd)
+	case "podman", "docker", "systemd":
+		return AddNonKubeSiteCommands(cmd)
+	default:
+		{
+			fmt.Printf("unsupported platform: %s\n", selectedPlatform)
+			return nil
+		}
+	}
+}
```

Comment:
@c-kruse this is the factory method to differentiate the creation of the kube/non-kube commands